src/js/main.js

```diff
@@ -0,0 +1,338 @@
+      } else if (event.target.classList.contains('main') || event.target.classList.contains('edition')) {
+        this.keysFilter(event.target.parentElement, true);
+      }
+      document.querySelector('.textarea').focus();
+    });
+
+    document.addEventListener('keydown', (event) => {
+      event.preventDefault();
+      if (event.repeat !== true || (event.repeat === true && event.shiftKey === true && event.key !== 'Shift')) {
+        if (this.keys[this.lang].map((item) => item[0]).indexOf(event.code) !== -1) {
+          document.querySelector(`.buttonClass[data-key=${event.code}]`).classList.add('act');
+          this.keysFilter(document.querySelector(`.buttonClass[data-key=${event.code}]`), false);
+        }
+      } else {
+        this.shift = true;
+      }
+      document.querySelector('.textarea').focus();
+    });
+
+    document.addEventListener('keyup', (event) => {
+
+      event.preventDefault();
+      if (this.keys[this.lang].map((item) => item[0]).indexOf(event.code) !== -1) {
+        document.querySelector(`.buttonClass[data-key=${event.code}]`).classList.remove('act');
+        this.keysFilter(document.querySelector(`.buttonClass[data-key=${event.code}]`), true);
+      }
+    });
+  }
+
+  keysFilter(value, up) {
+
+    const pattern = ['Backspace', 'ArrowLeft', 'ArrowRight',
+                      'ArrowDown', 'ArrowUp', 'Delete',
+                      'Tab', 'CapsLock', 'Enter',
+                      'ShiftLeft', 'ShiftRight', 'ControlLeft',
+                      'AltLeft', 'ControlRight', 'AltRight'];
+
+    if (pattern.indexOf(value.getAttribute('data-key')) === -1 && !up) {
+      this.CapsLock(value, true);
+    } else if (up) {
+      this.KeysUpper(value);
+    } else {
+      this.CapsLock(value, false);
+    }
+  }
+
+  CapsLock(value, print) {
+
+    if (!this.caps && print) {
+      this.printValue(value.querySelector('.main').innerHTML);
+    } else if (this.caps && print) {
+      if (value.querySelector('.edition')) {
+        this.printValue(value.querySelector('.edition').innerHTML);
+      } else {
+        this.printValue(value.querySelector('.main').innerHTML.toUpperCase());
+      }
+    } else {
+      this.KeyPress(value);
+    }
+  }
+
+  KeyPress(value) {
+
+    if (this.tmp.indexOf(value.querySelector('.main').innerHTML) === -1) {
+      this.tmp.push(value.querySelector('.main').innerHTML);
+    }
+
+    switch (value.querySelector('.main').innerHTML) {
+
+      case 'Backspace':
+        this.mouseID = document.querySelector('.textarea').selectionStart;
+        if (this.mouseID > 0) {
+
+          const memory = document.querySelector('.textarea').innerHTML;
+
+          this.tmpStr = memory.slice(0, this.mouseID - 1) + memory.slice(this.mouseID);
+          document.querySelector('.textarea').innerHTML = this.tmpStr;
+          document.querySelector('.textarea').selectionStart = this.mouseID - 1;
+          document.querySelector('.textarea').selectionEnd = this.mouseID - 1;
```

Comment:
Consider having `document.querySelector('.textarea')` as a constant to avoid copy-paste